### PR TITLE
Structure-based virtual screening methods and Treat TODOs

### PR DESCRIPTION
**Pull request by agitter**
This initial edit focuses on adding protein structure-based virtual screening methods to the treat section.  Before reviewing and merging, I'd like to address other TODOs in this section.

@rbharath does MoleculeNet's grid featurizer construct a 3D grid to represent the protein-ligand complex?  I wasn't sure whether to cite it in my line about 3D grid-based methods.

### CODE REVIEW

**Comment by agitter** [comment]
Given our time constraints, some TODOs no longer seen as important to me as they did originally.  I'm dropping some without addressing them.  Please let me know if I'm dropping something critical.

**Comment by agitter** [comment]
@cgreene in the Treat section you has suggested we link to a methodologically-related paper from Categorize:

> Gómez-Bombarelli et al. designed a  SMILES-to-SMILES autoencoder to learn a continuous latent feature space for chemicals [tag:Gomezb2016_automatic]. `TODO: connect to related EHR paper` In this learned continuous space it was possible to train some types of supervised learning algorithms or interpolate between continuous representations of chemicals in a manner that is not possible with discrete (e.g. bit vector or string) features.

Which paper was it?  And what should I add here?

**Comment by cgreene** [comment]
This doi is a paper by @brettbj : 10.1016/j.jbi.2016.10.007
In it, unsupervised pre-training of an autoencoder with large numbers of unlabeled records followed by training standard classifiers for a specific task dramatically reduce the number of examples required for successful classification. It's up to you as to whether or not that's relevant enough to include. Thanks!

**Comment by agitter** [comment]
Thanks for pointing me to the reference.  I'd say the main point of the paper here is that different types of supervised learning are possible in the continuous feature space learned by the autoencoder.  It's hard to interpolate between two discrete structures (e.g. strings).  It's easy to interpolate between two vectors in the latent space.

Does that seem relevant?

**Comment by cgreene** [comment]
I think that it's relevant to the review. It's probably something that would come naturally to the people working on this review but may not be obvious to the readers.

**Comment by agitter** [comment]
@cgreene I added a link in my latest commit.  What do you think?

**Comment by agitter** [comment]
I removed the [WIP] annotation, this is ready to be reviewed.  I originally planned on adding a longer introduction to Treat but @agapow's does a good job.

**Comment by rbharath** [comment]
The grid featurizer (as used in MoleculeNet) extracts a flat feature vector from the 3D protein-ligand structure. The featurizer can also extract a 3D grid representation, but we found this representation more unwieldy to work with than the flat feature vector.

**Comment by agitter** [comment]
Thanks for clarifying @rbharath.  My summary of tables 3 and 4 in your atomic convolutions paper is now:

> Gomes et al. demonstrate that currently random forest on a 1D feature vector that describes the 3D ligand-target structure generally outperforms neural networks on the same feature vector, atomic convolutions, and ligand-based neural networks when predicting the continuous-valued inhibition constant on the PDBBind refined dataset. However, in the long term, atomic convolutions may ultimately overtake grid-based methods, as they provide greater freedom to model atom-atom interactions and the forces that govern binding affinity.

File: sections/05_treat.md [hunk 1]
```diff
@@ -202,108 +193,110 @@ repositioning is currently missing.
...
+above [@doi:10.1021/acs.jproteome.6b00618].  An overly simplistic but
+approximately correct view of chemical fingerprints is that each bit represents
+the presence of absence of a particular chemical substructure in the molecular
+graph. Modern neural networks can operate directly on the molecular graph as
+input.  Duvenaud et al. [@tag:Duvenaud2015_graph_conv] generalized standard
+circular fingerprints by substituting discrete operations in the fingerprinting
+algorithm with operations in a neural network, producing a real-valued feature
+vector instead of a bit vector.  Other approaches offer trainable networks that
+can in theory learn chemical feature representations that are optimized for a
+particular prediction task.   Lusci et al. adapted recursive neural networks for
```
Comment (enricoferrero) [inline sections/05_treat.md]: Maybe move citation after "Lusci et al." for consistency

Comment (agitter) [inline sections/05_treat.md]: That makes sense.  Done.

**Comment by agitter** [comment]
The Travis CI builds are hanging, but I haven't made major changes since the last build passed.  I'll merge and check on the build status later.

**Pull request merged**